nixos/modules/services/misc/autorandr.nix

```diff
@@ -22,22 +22,16 @@ in {
 
     environment.systemPackages = [ pkgs.autorandr ];
 
-    # systemd.unitPackages = [ pkgs.autorandr ];
+    systemd.packages = [ pkgs.autorandr ];
+
     systemd.services.autorandr = {
-      unitConfig = {
-        Description = "autorandr execution hook";
-        After = [ "sleep.target" ];
-        StartLimitInterval = "5";
-        StartLimitBurst = "1";
-      };
-      serviceConfig = {
-        ExecStart = "${pkgs.autorandr}/bin/autorandr --batch --change --default default";
-        Type = "oneshot";
-        RemainAfterExit = false;
-      };
       wantedBy = [ "sleep.target" ];
     };
 
+    services.udev.extraRules = ''
+      ACTION=="change", SUBSYSTEM=="drm", RUN+="${pkgs.bash}/bin/sh -c '${pkgs.systemd}/bin/systemctl is-active -q graphical.target && ${pkgs.systemd}/bin/systemctl start autorandr.service'"
```

Comment:
@Mic92 
> set BindsTo to the device node to systemd to make the service restart

Sorry, can't get it. People use different devices. This is why the universal match of `SUBSYSTEM=="drm"` is used.